mockmp-runtime/src/jvmMain/kotlin/org/kodein/mock/platformJvm.kt

```diff
@@ -43,7 +43,23 @@ internal actual fun <T> KClass<*>.unsafeValue() = when (this) {
     Function8::class -> UnsafeFunctions.f8
     Function9::class -> UnsafeFunctions.f9
     Function10::class -> UnsafeFunctions.f10
-    else -> ObjenesisStd(true).newInstance(java)
+    else -> {
+        val target = if (this.java.isInterface) {
```

Comment:
This check isn't good enough to handle `sealed class`. We could check `constructors.size == 0` maybe, or even better, check the constructors like objenesis does.